guide/creating-your-bot/parsing-mention-arguments.md

```diff
@@ -0,0 +1,219 @@
+However, using `message.mentions` can lead to a few problems.  
+For example you do not really know which mention belongs to which argument.
+Or if you are splitting the message's content by spaces to get the args,
+then the mentions will still take up space in your args array which can mess up the rest of your args parsing if you are not careful.
+
+Say you are writing a bot for moderating your server. Obviously you will want a kick or a ban command which allows you to mention the person you are trying to ban.
+But what happens if you try to use the command like this?
+
+```
+!ban @Offender Because he was rude to @Victim
+```
+
+You might expect it to ban @Offender, because that is who you mentioned first.
+However, the Discord API does not send the mentions in the order they appear; They are sorted by their ID instead.
+
+If the @Victim happens to have joined Discord before @Offender and thus has a smaller ID he might get banned instead.  
+Or maybe someone uses a command incorrectly, the bot might still accept it but it will create an unexpected outcome.  
+Say someone accidentally used the ban command like this:
+
+```
+!ban Because he was rude to @Victim
+```
+
+The bot will still ban someone, but it will be the @Victim again. `message.mentions.users` still contains a mention, which the bot will use. But in reality you would want your bot to be able to tell the user he used the command incorrectly.
+
+### How Discord mentions work
+
+Discord uses a special syntax to embed mentions in a message. For user mentions it is the user's ID with `<@` at the start and `>` at the end, like this: `<@86890631690977280>`. If they have a nickname there will also be a be a `!` after the `@`.  
+Role mentions and channel mentions work similarly. Role mentions look like `<@&134362454976102401>` and channel mentions like `<#222197033908436994>`.
+
+That means when you receive a message from the Discord API and it contains mentions the message's content will contain that special syntax.  
+If you send
+
+<discord-messages>
+	<discord-message author="User" avatar="djs">
+		I think we should add @GoodPerson to the @Mod role.
+	</discord-message>
+</discord-messages>
+
+then the `message.content` for that message will look something like this
+
+<!-- eslint-skip -->
+```js
+'I think we should add <@86890631690977280> to the <@&134362454976102401> role.'
+```
+
+### Implementation
+
+So, how do you actually use this new information for your bot?  
+Most of your code will not change, however instead of using `message.mentions` to find the mentioned users you will have to do it manually.  
+This may sound scary at first, but once you see the code you will see it is actually pretty simple.
+
+Say you already have a simple command handler like this:
+
+```js
+client.on('message', message => {
+	// Don not go any further if there is no command prefix.
+	if (!message.content.startsWith(config.prefix)) return;
```

Comment:
I think explaining every single line of code, especially ones like this that are self-explanatory if you read them normally, is fairly redundant. Usually it reads better/feels less cluttered when you explain what the code is doing afterwards, outside of the codeblock. And even then, you can just give a general feel of it and not a line-by-line explanation.

Reply:
Though, as another note, I think since this is basically just skeleton code, it's doesn't necessarily need any explanation. The code that needs to be explained is mainly the mention parsing.